storage/types_test.go

```diff
@@ -26,6 +26,59 @@ import (
+		{val: 0x00, index: h2b("000102030405060708090A0B0C0D0E0F10111213"), depth: 159, want: h2b("000102030405060708090A0B0C0D0E0F10111212")},
+		{val: 0x00, index: h2b("000102030405060708090A0B0C0D0E0F10111213"), depth: 160, want: h2b("000102030405060708090A0B0C0D0E0F10111213")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 0, want: h2b("FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 1, want: h2b("7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 2, want: h2b("3FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 3, want: h2b("1FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 4, want: h2b("0FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 5, want: h2b("07FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 6, want: h2b("03FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 7, want: h2b("01FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 8, want: h2b("00FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 9, want: h2b("007FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 10, want: h2b("003FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 159, want: h2b("0000000000000000000000000000000000000001")},
+		{val: 0xFF, index: h2b("0000000000000000000000000000000000000000"), depth: 160, want: h2b("0000000000000000000000000000000000000000")},
+		{val: 0xFF, index: h2b("000102030405060708090A0B0C0D0E0F10111212"), depth: 1, want: h2b("7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("000102030405060708090A0B0C0D0E0F10111212"), depth: 17, want: h2b("00017FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")},
+		{val: 0xFF, index: h2b("000102030405060708090A0B0C0D0E0F10111212"), depth: 159, want: h2b("000102030405060708090A0B0C0D0E0F10111213")},
+		{val: 0xFF, index: h2b("000102030405060708090A0B0C0D0E0F10111212"), depth: 160, want: h2b("000102030405060708090A0B0C0D0E0F10111212")},
+	} {
+		nID := NewNodeIDFromHash(tc.index)
+		gotNode := nID.SetLowerBits(tc.depth, tc.val)
+		if got, want := gotNode.Path, tc.want; !bytes.Equal(got, want) {
+			t.Errorf("SetLowerBits(%x, %v, 0x%X): %x, want %x", tc.index, tc.depth, tc.val, got, want)
+		}
+		if got, want := gotNode.PrefixLenBits, 160; got != want {
```

Comment:
All the tests have bit length multiple of 8. Should we test the case when it's not?

Reply:
Since we are no longer manipulating the bit length, I don't think that's needed.  
`NewNodeIDFromHash` will always produce a bit length that is a multiple of 8.
Furthermore, `PrefixLenBits` is not used in the the computation.